pkgs/development/python-modules/pijuice/default.nix

```diff
@@ -0,0 +1,41 @@
+{ lib, fetchFromGitHub, buildPythonPackage, pythonOlder,
+smbus-cffi, urwid }:
+
+buildPythonPackage rec {
+  pname = "pijuice";
+  version = "1.7";
+  disabled = pythonOlder "3.6";
+
+  src = fetchFromGitHub {
+    owner = "PiSupply";
+    repo = "PiJuice";
+    # Unfortunately this project doesn't tag releases. This commit altered the
+    # __version__ field in pijuice.py
+    rev = "3ba6719ab614a3dc7495d5d9c900dd4ea977c7e3";
+    sha256 = "GoNN07YgVaktpeY5iYDbfpy5fxkU1x0V3Sb1hgGAQt4=";
+  };
+
+  patches = [
+    ./patch-shebang.diff
+  ];
+
+  PIJUICE_BUILD_BASE = 1;
+
+  sourceRoot = "source/Software/Source";
```

Comment:
```suggestion
  preBuild = ''
    cd Software/Source
  '';
```

Reply:
Is there a reason this should be used rather than `sourceRoot`?

Reply:
My personal reason is a subtle under-documented feature.

Usually, `fetchFromGitHub` sets the unpacked source file as `source` - more generally, whatever was set as `name` parameter for `fetchFromGitHub`. E.g. in the example below:

```nix
src = fetchFromGitHub {
   name = "foobar-sourceCode-xyz";
   owner = . . .;
   repo = . . .;
   . . .
};
```

the unpacked directory will be `foobar-sourceCode-xyz`.

This way, `sourceRoot` will be set to `foobar-sourceCode-xyz` in the example above.

Resetting `sourceRoot`, therefore, depends on the `name` attribute. If, by any reason, the fetcher changes this, then codes like `sourceRoot = "source/...";` will break.

--------------------------------

However, this is a "corner case". Do whatever you like.